### DISCUSSION TITLE
Which connector should I start with

### DISCUSSION THREAD

**Discussion by jimmarino** [International-Data-Spaces-Association/DataspaceConnector#53] [Q&A] [answered]
Hi,

I'm new to IDS and was wondering which dataspace connector I should start with. Specifically, I was interested in understanding whether I should be looking at this one or the trusted connector here:

https://github.com/industrial-data-space/trusted-connector

For example, do they serve different use cases such as highly secure environments vs more "open" data exchange? Basically, I'm trying to understand if there are plans to evolve both to serve different cases or there is one repo that will be developed moving forward.

Thanks,
Jim

**Comment by HeinrichPet** [comment, marked as answer]
Hi Jim,

thank you for this really good question.
Indeed, it depends on your use case. But it also depends on some technical things. In the end, you can decide which points are important for your company/project.

- The Trusted Connector is developed to get the Trust(+) [certification](https://www.internationaldataspaces.org/wp-content/uploads/2020/10/IDSA-White-Paper-Criteria-Catalogue-Components-Connector.pdf ), while the Dataspace Connector aims at the Base certification, but still includes additional criteria like usage control.
- The Dataspace Connector is officially validated to follow the Base certification criteria.
- The Dataspace Connector can run in Java, Docker or Kubernetes, while the Trusted Connector is based on Docker or [trust|me](https://github.com/trustm3/trustme_main) OS. Some of the security properties from the Trust(+) certification of the Trusted Connector are only supported if trust|me is used.
- The Trusted Connector requires specific hardware such as TPM or Intel SGX for secure remote attestation (a trust certification criteria).
- The Dataspace Connector uses https for data exchange in the IDS, whereas the Trusted Connector uses mainly [IDSCP](https://industrial-data-space.github.io/trusted-connector-documentation/docs/using_idscpv2/) for this. 
- Communication with the Dataspace Connector takes place via a defined http REST interface, while Camel routes must be created for the Trusted Connector.
- The Dataspace Connector uses Maven and Spring Framework. The Trusted Connector uses Gradle and Apache Karaf (OSGi).
Overall, the two developments are completely independent of each other, but still follow the same IDS architecture and IDS specifications. They differ mainly in their security features and used technologies.

Experience shows that the Trusted Connector is often used with Docker instead of trust|me, which means that the security properties from the Trust(+) certification are no longer fully supported. Developers also often find it difficult to use Apache Karaf (ODSGi) and Apache Camel in the Trusted Connector, but that's up to the experience of the developer. The Dataspace Connector is in that way easier to use, but more static by currently not supporting Apache Camel but a predefined REST interface. 

I hope that gives you a good overview about pros and cons to both of the implementations. I would be happy, if you ask more question like this :)

By the way, we currently build up a community around the Dataspace Connector. Therefore we explain the Dataspace Connector Roadmap and other points also in the [wiki](https://github.com/FraunhoferISST/DataspaceConnector/wiki). Feel free to bring up new topics, join our [Teams Meeting](https://www.dataspace-connector.io/community/) or contribute improvments to the Dataspace Connector.

Best regards,
Heinrich

**Comment by jimmarino** [comment]
Hi Heinrich,

Thanks for the detailed response, it was very helpful. I noticed the Teams meeting is set for every other week. Is there one today, Dec 18th?

Thanks!
Jim

**Reply by juliapampus** [reply]
Hi Jim, 

unfortunately, we do not have an appointment today. The next one would be on Tuesday 12.01.2021, at 11 a.m. 

Best regards, 
Julia

**Reply by jimmarino** [reply]
Hi Julia, Thanks. I'll attend the next meeting on the 12th.

**Reply by ghost** [reply]
welp
